libyul/optimiser/ForLoopCondRewriter.h

```diff
@@ -0,0 +1,38 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+#include <libyul/optimiser/ASTWalker.h>
+
+namespace yul
+{
+
+/**
+ * Rewrites ForLoop by moving the cond expression into the ForLoop.
+ * For example, `for {} lt(a, b) {}` becomes `for {} true { if iszero(lt(a, b)) }`
+ * Requirements:
+ * - The Disambiguator must be run upfront.
+ * - To avoid unnecessary rewrite, it is recommanded to run this rewriter after StructuralSimplifier.
+ */
+class ForLoopCondRewriter: public ASTModifier
```

Comment:
Since it would be good to also implement the reverse transform, perhaps name it `ForLoopConditionIntoBody`?

Reply:
What's the reverse transform's expected behavior? To find out all code piece similiar to `for { ... } 1 { ... } { if x { break } ... }` then turn it into `for { ... } iszero(x) { ... } { ... }`?

Should the reverse transform belongs to the scope of this pull request?

Reply:
Yes, something along those lines. The reverse transform can be a different pull request (and maybe should), but then you should not activate this transform in the suite.

Reply:
OK